presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -510,51 +523,52 @@ private void assertRoundTrip(Type writeType, Type readType, List<?> writeValues,
+        try (OrcRecordReader recordReader = createCustomOrcRecordReader(tempFile, orcEncoding, orcPredicate, types, MAX_BATCH_SIZE)) {
             assertEquals(recordReader.getReaderPosition(), 0);
             assertEquals(recordReader.getFilePosition(), 0);
 
             boolean isFirst = true;
             int rowsProcessed = 0;
-            Iterator<?> iterator = expectedValues.iterator();
             for (int batchSize = toIntExact(recordReader.nextBatch()); batchSize >= 0; batchSize = toIntExact(recordReader.nextBatch())) {
                 if (skipStripe && rowsProcessed < 10000) {
-                    assertEquals(advance(iterator, batchSize), batchSize);
+                    // skip recordReader.readBlock
                 }
                 else if (skipFirstBatch && isFirst) {
-                    assertEquals(advance(iterator, batchSize), batchSize);
+                    // skip recordReader.readBlock
                     isFirst = false;
                 }
                 else {
-                    Block block = recordReader.readBlock(type, 0);
-
-                    List<Object> data = new ArrayList<>(block.getPositionCount());
-                    for (int position = 0; position < block.getPositionCount(); position++) {
-                        data.add(type.getObjectValue(SESSION, block, position));
-                    }
-
-                    for (int i = 0; i < batchSize; i++) {
-                        assertTrue(iterator.hasNext());
-                        Object expected = iterator.next();
-                        Object actual = data.get(i);
-                        assertColumnValueEquals(type, actual, expected);
+                    for (int i = 0; i < types.size(); i++) {
+                        Type type = types.get(i);
+                        Block block = recordReader.readBlock(type, i);
+                        assertEquals(block.getPositionCount(), batchSize);
+
+                        List<Object> data = new ArrayList<>(block.getPositionCount());
+                        for (int position = 0; position < block.getPositionCount(); position++) {
+                            data.add(type.getObjectValue(SESSION, block, position));
+                        }
+
+                        for (int position = 0; position < block.getPositionCount(); position++) {
+                            assertColumnValueEquals(type, data.get(position), expectedValues.get(i).get(rowsProcessed + position));
+                        }
                     }
                 }
                 assertEquals(recordReader.getReaderPosition(), rowsProcessed);
                 assertEquals(recordReader.getFilePosition(), rowsProcessed);
                 rowsProcessed += batchSize;
             }
-            assertFalse(iterator.hasNext());
+            assertEquals(rowsProcessed, expectedValues.get(0).size());
```

Comment:
maybe
```java
int actualRowsProcessed = rowsProcessed;
expectedValues.forEach(value -> assertEquals(actualRowsProcessed, value.size()));

Reply:
I'll add a sanity check at the beginning of the method:

```
        int expectedRowsProcessed = expectedValues.get(0).size();
        expectedValues.forEach(values -> assertEquals(values.size(), expectedRowsProcessed));
```

and change this to

```
assertEquals(rowsProcessed, expectedRowsProcessed);
```